### ISSUE TITLE
Bug in release 1.64, DynamicParameterTest is failing

### ISSUE THREAD

**Issue by rvais** [cbeust/jcommander#329]
Build of the jcommander as a maven project will fail on DynamicParameterTest. Test method usage() fails with null pointer exception. When jcommander is trying to compare annotations of dynamic parameters in attempt to sort them, these annotations are null and due to method call exception is thrown. (Class com.beust.jcommander.JCommander$Options, parameterDescriptionComparator) Reproducer class is attached.
[BugReproducer.java.txt](https://github.com/cbeust/jcommander/files/777010/BugReproducer.java.txt)

**Comment by cbeust** [comment]
Your test is passing for me:

```
Usage: <main class> [options]
  Options:
    -A

      Syntax: -Akey@value
      Default: {}
    -D
      Dynamic parameters go here
      Syntax: -Dkey=value
      Default: {}
```

Do you have a stack trace?

**Comment by rvais** [comment]
I have stack trace and maven output (both follows below). If necessary, I can provide pom.xml file used. It is important to mention that if I use recommended method of building jcommander (./kobaltw assemble) everything seems to work. If I can provide more information, please let me  know.

stack trace for reproducer class:
```
--- exec-maven-plugin:1.2.1:exec (default-cli) @ jcommander ---
Exception in thread "main" java.lang.NullPointerException
	at com.beust.jcommander.JCommander$Options$1.compare(JCommander.java:144)
	at com.beust.jcommander.JCommander$Options$1.compare(JCommander.java:139)
	at java.util.TimSort.countRunAndMakeAscending(TimSort.java:355)
	at java.util.TimSort.sort(TimSort.java:220)
	at java.util.Arrays.sort(Arrays.java:1512)
	at java.util.ArrayList.sort(ArrayList.java:1454)
	at java.util.Collections.sort(Collections.java:175)
	at com.beust.jcommander.JCommander.usage(JCommander.java:1158)
	at com.beust.jcommander.JCommander.usage(JCommander.java:1113)
	at com.beust.jcommander.BugReproducer.main(BugReproducer.java:39)
------------------------------------------------------------------------
BUILD FAILURE
------------------------------------------------------------------------
```

part of maven output:
```
[rvais]$ mvn verify

...

-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running TestSuite
Tests run: 200, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 1.702 sec <<< FAILURE!

Results :

Failed tests:   usage(com.beust.jcommander.dynamic.DynamicParameterTest)

Tests run: 200, Failures: 1, Errors: 0, Skipped: 0
```

**Comment by cbeust** [comment]
The stack trace doesn't match the source:

https://github.com/cbeust/jcommander/blob/master/src/main/java/com/beust/jcommander/JCommander.java#L144

Are you sure you're using the latest version?

Again, just give me a simple test case that causes this NPE, that's all I need.

**Comment by rvais** [comment]
Well for master branch it does work. But as title suggest I am not concerned about master branche. I am concerned about release 1.64. I work as java packager for Fedora at Red Hat and I was supposed to report this as a bug. I can't package this release if it won't passes the test.
[https://github.com/cbeust/jcommander/blob/1.64/src/main/java/com/beust/jcommander/JCommander.java#L144](url)

**Comment by cbeust** [comment]
Something is weird, I'm not seeing the source I expect to see.

In particular, this commit:

https://github.com/cbeust/jcommander/commit/6a719a1bc9317ce1ce14cfd2e3cf018ef90dfa6b

can't be seen in the commit list:

https://github.com/cbeust/jcommander/commits/master

And I don't remember having made it in a branch either, it's all happening in `master`.

I'm very confused right now.

**Comment by cbeust** [comment]
@rvais I'm curious, how did you find that commit at all? I don't see it in the history (but the commit is real and it should be in the code base, I'm baffled why I can't find it).

**Comment by cbeust** [comment]
I still don't understand where that commit is but I just published 1.65,
which should fix the bug.


-- 
Cédric

**Comment by mizdebsk** [comment]
> I'm curious, how did you find that commit at all?

@cbeust You can go to [Releases](https://github.com/cbeust/jcommander/releases) and click on 1.64

> I still don't understand where that commit is but I just published 1.65, which should fix the bug.

Thank you.

**Comment by cbeust** [comment]
It's still a mystery to me why that commit doesn't appear in the commit log, but well.

Closing.
